frontend/utils/api.js

```diff
@@ -0,0 +1,100 @@
+const API_URL = process.env.NEXT_PUBLIC_API_URL;
+
+export const fetchMeals = async (
+  query = "",
+  sortKey = "scheduled_at",
+  sortDir = "asc",
+  availableReservations = false
+) => {
+  const url = new URL(`${process.env.NEXT_PUBLIC_API_URL}/meals/summary`);
+
+  if (query) {
+    url.searchParams.append("title", query);
+  }
+
+  if (availableReservations) {
+    url.searchParams.append("availableReservations", "true");
+  }
+
+  url.searchParams.append("sortKey", sortKey);
+  url.searchParams.append("sortDir", sortDir);
+
+  const response = await fetch(url, {
+    cache: "no-store",
+  });
+
+  if (!response.ok) {
+    throw new Error("Failed to fetch meals");
+  }
+
+  return response.json();
+};
+
+export const fetchMeal = async (id, setMeal, setError, setLoading) => {
+  try {
+    const [mealResponse, reviewsResponse, reservationsResponse] =
+      await Promise.all([
+        fetch(`${API_URL}/meals/${id}`),
+        fetch(`${API_URL}/meals/${id}/reviews`),
+        fetch(`${API_URL}/meals/${id}/reservations`),
+      ]);
+
+    if (!mealResponse.ok || !reviewsResponse.ok || !reservationsResponse.ok)
+      throw new Error("Failed to fetch information about this meal");
```

Comment:
running api calls in parallel is useful in the case where the different calls can be slow. In this case it is not really necessary and will it make it harder for you to debug, so here i recommend you split up the calls. Also if the first one fails, you wont need to call the other ones.

Reply:
Multiple calls also have the problem that the meal might change in between the different calls. E.g. you could fetch the meal, then another client deletes the meal and then your two next calls would fail.

Look into graphql, where you can specify exactly the data you want

Reply:
It makes a lot of sense. I have split up the calls. 
And I agree; GraphQL is definitely on my learning list.